tests/manager/api/controller/test_odl_notify.py

```diff
@@ -1,132 +1,147 @@
-            response = controller_fixture.manager.odl_notification_controller.notify(
-                loan.id
-            )
-            assert 200 == response.status_code
-
-            # Update loan was called with the expected arguments.
-            update_loan_mock.assert_called_once_with(loan, json.loads(text))
-
-    def test_notify_errors(self, controller_fixture: ControllerFixture):
-        db = controller_fixture.db
-
-        # No loan.
-        with controller_fixture.request_context_with_library("/", method="POST"):
-            response = controller_fixture.manager.odl_notification_controller.notify(
-                db.fresh_str()
-            )
-            assert NO_ACTIVE_LOAN.uri == response.uri
+        assert odl_fixture.license.checkouts_available == 0
 
-        # Loan from a non-ODL collection.
+        status_doc = odl_fixture.loan_status_document("revoked")
+        with flask_app_fixture.test_request_context(
+            "/",
+            method="POST",
+            data=status_doc.model_dump_json(),
+            library=odl_fixture.library,
+        ):
+            assert loan.id is not None
+            response = odl_fixture.controller.notify(loan.id)
+            assert response.status_code == 204
+
+        assert odl_fixture.license.checkouts_available == 1
+
+    def test_notify_errors(
+        self,
+        db: DatabaseTransactionFixture,
+        flask_app_fixture: FlaskAppFixture,
+        odl_fixture: ODLFixture,
+    ):
+        # Bad JSON.
         patron = db.patron()
         pool = db.licensepool(None)
         loan, ignore = pool.loan_to(patron)
         loan.external_identifier = db.fresh_str()
+        with flask_app_fixture.test_request_context(
+            "/", method="POST", library=odl_fixture.library
+        ):
+            response = odl_fixture.controller.notify(loan.id)
+        assert response == INVALID_INPUT
 
-        with controller_fixture.request_context_with_library("/", method="POST"):
-            response = controller_fixture.manager.odl_notification_controller.notify(
-                loan.id
-            )
-            assert INVALID_LOAN_FOR_ODL_NOTIFICATION == response
+        # Loan from a non-ODL collection.
+        with flask_app_fixture.test_request_context(
+            "/",
+            method="POST",
+            library=odl_fixture.library,
+            data=odl_fixture.loan_status_document("active").model_dump_json(),
+        ):
+            response = odl_fixture.controller.notify(loan.id)
+        assert isinstance(response, ProblemDetail)
+        assert response == INVALID_LOAN_FOR_ODL_NOTIFICATION
+
+        # No loan, but distributor thinks it isn't active
+        with flask_app_fixture.test_request_context(
+            "/",
+            method="POST",
+            library=odl_fixture.library,
+            data=odl_fixture.loan_status_document("returned").model_dump_json(),
+        ):
+            response = odl_fixture.controller.notify(-55)
```

Comment:
I think this `-55` is meant to represent a loan that can't exist, but it's not super clear in this context. Would be useful to add a comment here (and a few lines down) making that explicit. Or we could use a constant named something like "NON_EXISTENT_LOAN".